articles/File_download.md

```diff
@@ -0,0 +1,60 @@
+# Download file from a website and upload it to Dropbox 
+
+## My very first function in powershell ~
+
+```powershell
+function Get-UrlInfo {
+        [Cmdletbinding()]
+        param(
+                [Parameter(Mandatory=$true)]
+                [String]$fileName
+
+        )
+        $fileContent = Get-Content $fileName
+        $fileString = $fileContent.Split("</p>")[1] + "</p>"
+        $xmlHeader= [xml]$fileString
+        $hrefString = $xmlHeader.p.a.href
+        $key = ([System.Uri]$hrefString).Query.Split("=")[1]
+        $url = "https://fast.wistia.net/embed/iframe/" + $key
+
+        # Download the video from the above URL
+        youtube-dl $url
+
+        # Get the name of the downloaded file
+        $outputFile = Get-ChildItem *.bin -Name
+        $fileString = $outputFile.Split("-")[0]+"-$key" + ".mp4"
```

Comment:
We could probably extract this as a separate function and simply call that function here.

Reply:
Ok. I have to try this.

Reply:
How do I extract a separate function for this, as I have to rely on `$key` variable, which means I will have to execute all the above steps first? I wonder how creating a separate function will make any difference here. What do you think?

Reply:
Well, think of this as a separate box. Perhaps, it's best to discuss this on call :)